components/style/properties/longhand/inherited_text.mako.rs

```diff
@@ -936,6 +936,111 @@ ${helpers.single_keyword("text-align-last",
     }
 </%helpers:longhand>
 
+<%helpers:longhand name="text-emphasis-position" animatable="False" products="none">
+    use std::fmt;
+    use values::computed::ComputedValueAsSpecified;
+    use values::NoViewportPercentage;
+    use style_traits::ToCss;
+
+    #[derive(Debug, Clone, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub enum VerticalValue {
+        Right,
+        Left
+    }
+
+    #[derive(Debug, Clone, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub enum HorizontalValue {
+        Over,
+        Under
+    }
+
+    #[derive(Debug, Clone, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub enum SpecifiedValue {
+        Value(HorizontalValue, VerticalValue)
+    }
+
+    pub mod computed_value {
+        pub type T = super::SpecifiedValue;
+    }
+
+    impl ComputedValueAsSpecified for SpecifiedValue {}
+    impl NoViewportPercentage for SpecifiedValue {}
+
+    pub fn get_initial_value() -> computed_value::T {
+        SpecifiedValue::Value(HorizontalValue::Over, VerticalValue::Right)
+    }
+
+    pub fn parse(_context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
```

Comment:
If you're interested in experimenting on something new, then [there's a new type](https://github.com/servo/servo/blob/9564673b5a1c5d078fd738abd334d201f9d228f0/components/style/values/mod.rs#L102) we'd introduced lately for CSS types which can take one of two values. All you have to do is define `right`, `left`, etc. [like so](https://github.com/servo/servo/blob/9564673b5a1c5d078fd738abd334d201f9d228f0/components/style/values/mod.rs#L96) and use something like,

``` rust
pub type VerticalValue = Either<Left, Right>;
pub type HorizontalValue = Either<Over, Under>;
pub type T = Either<HorizontalValue, VerticalValue>;
```

The advantage of this is that you don't have to write a lot of manual parsing/serialization code - just the necessary ones.